\name{scoreClinVarQSW}
\alias{scoreClinVarQSW}
%- Also NEED an '\alias' for EACH other topic documented here.
\title{
%%  ~~function to do ... ~~
  ClinVar-quantiled SeqWeaver scores
}
\description{
%%  ~~ A concise (1-5 lines) description of what the function does. ~~
  Transform SeqWeaver scores to ClinVar-quantiled SeqWeaver scores based on the empircal quantiles.
}
\usage{
  scoreClinVarQSW(variants, SW, target, empirical=0)
}
%- maybe also 'usage' for other objects documented here.
\arguments{
  \item{variants}{
%%     ~~Describe \code{variants} here~~
   A data frame of variants containing columns of \var{Chr}, \var{Start}, and \var{End}.
}
  \item{SW}{
%%     ~~Describe \code{SW} here~~
  A data frame of SeqWeaver scores, sourced from the output file generated by the SeqWeaver software (typically found at ".../final_outputs/seqweaver_results.tsv").
}
  \item{target}{
%%     ~~Describe \code{target} here~~
  A character string of RNA-binding protein (RBP). The columns with column names containing \code{target} will be extracted from \code{SW}.
}
  \item{empirical}{
%%     ~~Describe \code{empirical} here~~
  Either a numerical value or a data frame of user-defined quantiles of SeqWeaver scores.
  \enumerate{
    \item If \code{empirical} is a numerical value, the pre-computed quantiles of SeqWeaver scores using ClinVar single-nucleotide variants (SNVs) will be used. \code{empirical} = 0 indicates the use of pre-computed quantiles from all SNVs, and \code{empirical} = 1 indicates the use of pre-computed quantiles from pathogenic SNVs only.
    \item If \code{empirical} is a data frame, it should contain (1) a \var{Quantile} column indicating the proportion of values less than the given value and (2) columns of quantiles with column names consistent with the corresponding column names in \code{SW}.
  }
}
}
\details{
%%  ~~ If necessary, more details than the description above ~~
SeqWeaver scores for a variant are transformed to represent the proportion of ClinVar single-nucleotide variants (SNVs) with scores lower than that of the variant.  When multiple SeqWeaver scores exist for the same RNA-binding protein (\code{target}) based on mutliple trained SeqWeaver models of that RBP (\code{target}), the maximum score is reported as Clinvar-quantiled SeqWeaver scores.

This transformation is achieved by comparing the SeqWeaver scores to pre-computed empirical quantiles derived from SeqWeaver scores of ClinVar SNVs. These empirical quantiles from ClinVar are only available for SeqWeaver models trained on eCLIP-seq data of RBPs in HepG2 and K562 cell lines. If needed, pre-computed empirical quantiles can be substituted with other estimated quantiles provided in \code{empirical}.

}
\value{
%%  ~Describe the value returned
%%  If it is a LIST, use
%%  \item{comp1 }{Description of 'comp1'}
%%  \item{comp2 }{Description of 'comp2'}
%% ...
A new data frame incorporating the contents of \code{variants} along with the following changes:
  \itemize{
    \item Columns of SeqWeaver scores for the given \code{target} are extracted from \code{SW} and added to \code{variants}.
    \item \var{QSWscore} representing ClinVar-quantiled SeqWeaver scores computed using either empirical quantiles from ClinVar SNVs or user-defined quantiles in \code{empirical} is added to \code{variants}.
    \item \var{QSWscore} is added to \var{INCAscore}. If \code{variants} does not already contain \var{INCAscore}, \var{INCAscore} is added and set to 0 for all variants initially.
  }
}
\references{
%% ~put references to the literature/web site here ~
}
\author{
%%  ~~who you are~~
Jingqi Duan
}
\note{
%%  ~~further notes~~
}

%% ~Make other sections like Warning with \section{Warning }{....} ~

\seealso{
%% ~~objects to See Also as \code{\link{help}}, ~~~
}
\examples{
directory = 'https://raw.github.com/jduan607/INCA/master'
variants = fread(file.path(directory, 'GWAS', 'final_data.txt'))
SW = fread(file.path(directory,'GWAS','seqweaver_results.tsv'))

# Empirical quantiles of SeqWeaver scores from ClinVar SNVs
variants1 = scoreClinVarQSW(variants, SW, 'HNRNPK', empirical=1)

# Estimated (user-defined) quantiles of SeqWeaver scores
# This is for example purposes only, and the true distribution may be different.
probs = seq(0,1,by=.001)
qdata = data.frame(probs, qnorm(probs, mean=0, sd=1),
                   qnorm(probs, mean=3, sd=1),
                   qexp(probs, rate = 1),
                   qexp(probs, rate = 1) + 2)
colnames(qdata) = c('Quantile', grep('HNRNPK', colnames(SW), value=TRUE))
variants2 = scoreClinVarQSW(variants, SW, 'HNRNPK', empirical=qdata)
}
% Add one or more standard keywords, see file 'KEYWORDS' in the
% R documentation directory (show via RShowDoc("KEYWORDS")):
% \keyword{ ~kwd1 }
% \keyword{ ~kwd2 }
% Use only one keyword per line.
% For non-standard keywords, use \concept instead of \keyword:
% \concept{ ~cpt1 }
% \concept{ ~cpt2 }
% Use only one concept per line.
